sympy/solvers/tests/test_solvers.py

```diff
@@ -1896,6 +1896,17 @@ def test_lambert_bivariate():
     assert solve((x**3)**(x/2) + pi/2, x) == [
         exp(LambertW(-2*log(2)/3 + 2*log(pi)/3 + I*pi*Rational(2, 3)))]
 
+    # issue 23253
+    assert solve((1/log(sqrt(x) + 2)**2 - 1/x)) == [
```

Comment:
Does `solve((1/log(1/sqrt(x) + 2)**2 - x))` also work and give `[(LambertW(-exp(-2), -1) + 2)**-2`]?